### PR TITLE
Update Ledger API's proto documentation [KVL-1094]

### PR DESCRIPTION
**Pull request by miklos-da**
Summary of changes:
* Updated proto documentation to reflect changes primarily related to command deduplication (mostly taken from #9567).

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 1]
```diff
@@ -71,6 +75,8 @@ message Commands {
 
       // Specifies the length of the deduplication period.
       // It is interpreted relative to the local clock at some point during the submission's processing.
+      // If omitted, the participant will assume the configured maximum deduplication time (see
+      // ``ledger_configuration_service.proto``).
```
Comment (fabiotudone-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:79]: 👍🏼

**Review by fabiotudone-da — COMMENTED** [review]
Could you also make it explicit in the command service proto that completions may also be produced?

**Comment by miklos-da** [comment]
> Could you also make it explicit in the command service proto that completions may also be produced?

Sure, added a note to `command_service.proto` — let me know if that's what you had in mind.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto [hunk 2]
```diff
@@ -15,6 +15,9 @@ option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 
 // Command Service is able to correlate submitted commands with completion data, identify timeouts, and return contextual
 // information with each tracking result. This supports the implementation of stateless clients.
+//
+// Note that submitted commands MAY produce a completion event as well. For example, the participant MAY choose to
```
Comment (tudor-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto:19]: I don't understand what this sentence brings. Submitted commands usually lead to completion events

Comment (fabiotudone-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto:19]: This was explicitly stated for the command submission service but not for the command service (where it's less obvious, since it's synchronous).

**Review by fabiotudone-da — APPROVED** [review]

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto (hunk 2 above)
Comment (tudor-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto:19]: The use of that capitalized **MAY** suggests to me that completion events stemmed from submitted commands via the CommandService are the exception, not the norm. And the follow-up sentence doesn't clarify the idea. Maybe rewording it, if I understood the purpose right, might help: `Note that submitted commands produce completion events as well, even in some cases when commands get rejected (e.g. the participant MAY choose to produce a completion event for a rejection of a duplicate command)`

Comment (miklos-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto]: @tudor-da Reworded accordingly — let me know what you think.

Comment (tudor-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto]: Thanks lgtm 👍

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 3]
```diff
@@ -34,9 +33,10 @@ message Commands {
   // Required
   string application_id = 3;
 
-  // Uniquely identified the command. This identifier should be unique for each new command within an
-  // application domain, i.e., the triple (application_id, party, command_id) must be unique.
-  // It can be used for matching the requests with their respective completions.
+  // Uniquely identifies the command.
+  // The triple (application_id, party + act_as, command_id) constitutes the change ID for the intended ledger change,
+  // where party + act_as is interpreted as a set of party names.
+  // The change ID can be used for matching the intended ledger changes with their respective completions.
```
Comment (andreaslochbihler-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:39]: Isn't this what the `submission_id` is good for? Since there can be multiple completions for the same change ID (e.g., if there's a completion that says that the command already exists).

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 4]
```diff
@@ -57,20 +57,27 @@ message Commands {
...
+
+    // Specifies the start of the deduplication period by a completion stream offset.
+    //
+    // Must be a valid LedgerString (as described in ``value.proto``).
+    string deduplication_offset = 15;
+
+    // Specifies the length of the deduplication period.
+    // It is interpreted relative to the local clock at some point during the submission's processing.
+    // If omitted, the participant will assume the configured maximum deduplication time (see
+    // ``ledger_configuration_service.proto``).
```
Comment (andreaslochbihler-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:77]: Since this sits in a `oneof`, wouldn't it make more sense to move this to line 61.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto [hunk 5]
```diff
@@ -57,20 +57,27 @@ message Commands {
   // Required
   repeated Command commands = 8;
 
+  // Specifies the deduplication period for the change ID.
+  // If omitted, the participant chooses a deduplication period at its discretion.
```
Comment (andreaslochbihler-da) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:61]: This is somewhat at odds with what is now written for `deduplication_duration`.

File: ledger/ledger-configuration/protobuf/com/daml/ledger/configuration/ledger_configuration.proto [hunk 6]
```diff
@@ -32,7 +32,7 @@ message LedgerConfiguration {
   // ledger effective time and maximum record time of transactions.
   LedgerTimeModel time_model = 3;
 
-  // The maximum value for the ``deduplication_time`` parameter of command submissions
+  // The maximum value for the ``deduplication_duration`` parameter of command submissions
```
Comment (andreaslochbihler-da) [inline ledger/ledger-configuration/protobuf/com/daml/ledger/configuration/ledger_configuration.proto:35]: This is a bit add odds with the new description in the `LedgerConfiguration`:
```
Command submissions with longer periods MAY get accepted though.
```
Can you adjust it to match what you wrote in `ledger_configuration_service.proto`?

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto (hunk 5 above)
Comment (miklos-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:61]: Replaced it with the description originally next to `deduplication_duration`.

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto (hunk 3 above)
Comment (miklos-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:39]: You're right, slightly changed the wording to emphasize the change ID can be used for finding all relevant completions.

Comment (miklos-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto:39]: Addressed in #10851.

File: ledger/ledger-configuration/protobuf/com/daml/ledger/configuration/ledger_configuration.proto (hunk 6 above)
Comment (miklos-da) [reply ledger/ledger-configuration/protobuf/com/daml/ledger/configuration/ledger_configuration.proto:35]: Addressed in #10851 -- let me know what you think.
